pkg/registry/core/pod/storage/eviction.go

```diff
@@ -325,12 +326,22 @@ func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string,
 		}
 
 		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, conditionAppender)
+		podCopyUpdatedWithDeletePreconditions := &updatedObjectInfoWithOverridePreconditions{podCopyUpdated, deleteOptions.Preconditions}
 
-		if _, _, err = r.store.Update(ctx, name, podCopyUpdated, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{}); err != nil {
+		updatedPodObject, _, err := r.store.Update(ctx, name, podCopyUpdatedWithDeletePreconditions, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{DryRun: deleteOptions.DryRun})
```

Comment:
oof... this means we were actually updating the pod with a condition as part of handling a dry-run evict previously? I'm not sure there's any point in adding the condition if we're in dry run mode, since the whole point of that is to be observed by other API callers

Reply:
yeah, not great.. Ok, I suppose we can skip the whole block to simplify the execution.